compiler/accelerators/snax_gemm.py

```diff
@@ -0,0 +1,185 @@
+                "strideInnermostB": 0x3C5,
+                "strideInnermostC": 0x3C6,
+                "ldA": 0x3C7,
+                "ldB": 0x3C8,
+                "ldC": 0x3C9,
+                "strideA": 0x3CA,
+                "strideB": 0x3CB,
+                "strideC": 0x3CC,
+                "subtractions": 0x3CE,
+            },
+            {"launch": 0x3CF},
+            0x3CF,
+        )
+
+    def convert_to_acc_ops(self, op: linalg.Generic) -> Sequence[Operation]:
+        """
+        Lowers the operation op to a sequence of acc_ops.
+        acc_ops are:
+            - *.op that generates SSAValues consumed by acc2.setup
+            - acc2.setup
+            - acc2.launch
+            - acc2.await
+        These ops can further be lowered by specific instances of the
+        Accelerator interface
+        """
+        args = self._generate_setup_vals(op)
+
+        ops_to_insert = []
+        # insert ops to calculate arguments
+        for new_ops, _ in args:
+            ops_to_insert.extend(new_ops)
+
+        return [
+            *ops_to_insert,
+            setup := accfg.SetupOp([val for _, val in args], self.fields, self.name),
+            launch_val := arith.Constant(builtin.IntegerAttr.from_int_and_width(1, 5)),
+            token := accfg.LaunchOp([launch_val], self.launch_fields, setup),
+            accfg.AwaitOp(token),
+        ]
+
+    def _generate_setup_vals(
+        self, op: linalg.Generic
+    ) -> Sequence[tuple[Sequence[Operation], SSAValue]]:
+        """
+        Produce a `Sequence[Operation], SSAValue` tuple
+        for each field that contains:
+
+        - a list of operations that calculate the field value
+        - a reference to the SSAValue containing the calculated field value
+        """
+
+        def _generate_subtract_config(subtraction_a: SSAValue, subtraction_b: SSAValue):
+            """
+            Helper function to calculate a subtraction configuration for SNAX GEMM
+
+            Should be equivalent to C code:
+
+            int32_t gen_subtraction_config(int8_t subtraction_a, int8_t subtraction_b) {
+                return ((uint8_t)subtraction_b << 8) | (uint8_t)subtraction_a;
+            }
+
+            Instead of casting to 8-bit unsinged integer, we use AND with 0xFF.
+            This prevents a possible overflow on shifting the values.
+            """
+            return [
+                c_8 := arith.Constant.from_int_and_width(8, 32),
+                c_0xFF := arith.Constant.from_int_and_width(0xFF, 32),
+                subtraction_a_no_overflow := arith.AndI(subtraction_a, c_0xFF),
+                subtraction_b_no_overflow := arith.AndI(subtraction_b, c_0xFF),
+                shifted_subtraction_b := arith.ShLI(subtraction_b_no_overflow, c_8),
+                or_op := arith.OrI(subtraction_a_no_overflow, shifted_subtraction_b),
+            ], or_op.result
+
+        def _generate_size_config(Batch: int, M: int, K: int, N: int):
```

Comment:
nit: `Batch` should be lower case

Reply:
All of them should probably be lower case

Reply:
Okay!